### PR TITLE
Replace `lodash.sortby` and fix sources table ordering by url

### LINKED ISSUE #1108: Replace `lodash.sortby` with `Array.prototype.sort`
## Problem

We are looking to remove dependency on lodash altogether (see milestone description).

## Description

Replace all usages of `lodash.sortby` with [`Array.prototype.sort`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort).

Please note that `Array.prototype.sort` is not a pure function and mutates the array. Be sure to copy the array before sorting it with native sort.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #1388: Domain sorting in sources table doesn't take the clean URL
## Description

When sorting the table by the **Domain** field it doesn't appear like it's sorted because the `lodash.sortby` function uses the raw URL instead of the cleaned version, which is the one shown.

## Reproduction

1. Go to the /sources page (in wp.org/sources, [staging](https://search-staging.openverse.engineering/sources) or local)
2. Click on the "**Domain**" header cell of the table
3. See "clevelandart.org " after "postalmuseum.si.edu" among other misplaced URLs.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #1108 by @sarayourfriend 
Fixes #1388 by @krysal

## Description

This PR completes the **Remove lodash** milestone by replacing `lodash.sortby` with `Array.prototype.sort` as Sara suggested. There is still a `lodash.isempty` dependency that I'm not sure if it can be safely removed, doesn't seem to be used anywhere.
 
Also fixes the sorting of the Sources table, something I noticed while working in the first issue :)

## Testing Instructions

Reproduce steps in #1388 and see URLs correctly sorted.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VSourcesTable.vue [hunk 1]
```diff
@@ -110,15 +111,28 @@ export default defineComponent({
...
       return removeAfterSlash(stripLeadingWww(stripProtocol(url)))
     }
+
     const getLocaleFormattedNumber = useGetLocaleFormattedNumber()
     const providerStore = useProviderStore()
 
+    function compareProviders(prov1: MediaProvider, prov2: MediaProvider) {
+      let field1 = prov1[sorting.field]
+      let field2 = prov2[sorting.field]
+      if (sorting.field == 'source_url') {
```
Comment (zackkrida) [inline src/components/VSourcesTable.vue:121]:
```suggestion
      if (sorting.field === 'source_url') {
```

File: src/components/VSourcesTable.vue [hunk 2]
```diff
@@ -90,10 +90,10 @@ export default defineComponent({
   setup(props) {
     const sorting = reactive({
       direction: 'asc',
-      field: 'display_name',
+      field: 'display_name' as keyof Omit<MediaProvider, 'logo_url'>,
```
Comment (obulat) [inline src/components/VSourcesTable.vue:93]: I wonder if we should remove the provider `logo_url` parameter. Do we use it anywhere?

**Review by obulat — COMMENTED** [review]
The bundle size change is 🔥
I will try to approve this tomorrow, after running and making sure it works.

File: package.json [hunk 3]
```diff
@@ -84,7 +84,6 @@
     "focus-visible": "^5.2.0",
     "glob": "^7.1.6",
     "lodash.isempty": "^4.4.0",
```
Comment (sarayourfriend) [inline package.json:86]: It should be safe to remove this! It's not explicitly imported by anything. If it's needed by another dependency it'll still get included in the lock file. It's not hurting our bundle for it to be here, but someone might accidentally use it :stuck_out_tongue:

**Review by sarayourfriend — APPROVED** [review]
LGTM! Works perfect :tada: 

Once this is merged we can close the "remove lodash" milestone! Congrats @krysal!

**Review by zackkrida — APPROVED** [review]
LGTM!

**Review by obulat — APPROVED** [review]

File: src/components/VSourcesTable.vue (hunk 2 above)
Comment (krysal) [reply src/components/VSourcesTable.vue:93]: No, I don't think we are showing the sources logo anywhere, though it was the intention in CC to show them on the homepage if I recall correctly? we can probably remove them!

**Pull request merged**